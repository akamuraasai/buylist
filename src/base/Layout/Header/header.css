.Header {
    width: 100%;
    height: 120px;
    display: inline-flex;
    background-color: #76b43b;
    color: #f7faf5;
    align-items: center;
    justify-content: center;
    margin-top: 60px;
}

.Header__value {
    display: flex;
    height: 120px;
    width: 60%;
    align-items: center;
    justify-content: center;
}

.Header__icon {
    font-size: 60px;
    width: 80px;
    text-align: center;
    align-items: center;
    justify-content: center;
}

.Header__value__text {
    width: 130px;
    padding-top: 10px;
    text-align: left;
    align-items: center;
    justify-content: center;
}

.Header__value__text p {
    margin: 0;
    font-size: 18px;
}