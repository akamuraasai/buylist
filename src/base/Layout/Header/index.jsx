import React from 'react';
import PropTypes from 'prop-types';

import './header.css';
import moneyFormatter from '../../../helpers/moneyFormatter';

const calculateTotal = items => items.reduce((total, item) => total + (item.price * item.units), 0);
const toPrice = moneyFormatter;

const Header = ({ products }) => (
  <div className="Header">
    <div className="Header__value">
      <div className="Header__icon">
        <i className="fa fa-shopping-cart"></i>
      </div>
      <div className="Header__value__text">
        <p>Total</p>
        <p>R$ {toPrice(calculateTotal(products))}</p>
      </div>
    </div>
  </div>
);

Header.defaultProps = {
  products: [],
};

Header.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape()),
};

export default Header;
