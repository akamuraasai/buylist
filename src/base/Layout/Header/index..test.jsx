import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';

describe('Header', () => {
  const header = shallow(<Header />);

  it('should render without crashing', () => {
    expect(header).toMatchSnapshot();
  });

  it('should render the total price', () => {
    const items = [
      { price: 5, units: 1 },
      { price: 4, units: 10 },
    ];
    const headerWithItems = shallow(<Header products={items} />);

    expect(headerWithItems.find('.Header__value__text > p').at(1).text()).toEqual('R$ 45,00');
  });
});
