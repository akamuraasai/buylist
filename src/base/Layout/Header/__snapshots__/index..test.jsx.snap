// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    products={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Header__value"
>
        <div
                className="Header__icon"
        >
                <i
                        className="fa fa-shopping-cart"
                />
        </div>
        <div
                className="Header__value__text"
        >
                <p>
                        Total
                </p>
                <p>
                        R$ 
                        0,00
                </p>
        </div>
</div>,
      "className": "Header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Header__icon"
>
            <i
                        className="fa fa-shopping-cart"
            />
</div>,
          <div
            className="Header__value__text"
>
            <p>
                        Total
            </p>
            <p>
                        R$ 
                        0,00
            </p>
</div>,
        ],
        "className": "Header__value",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-shopping-cart"
/>,
            "className": "Header__icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-shopping-cart",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Total
</p>,
              <p>
                R$ 
                0,00
</p>,
            ],
            "className": "Header__value__text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total",
              },
              "ref": null,
              "rendered": "Total",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "R$ ",
                  "0,00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "R$ ",
                "0,00",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Header__value"
>
          <div
                    className="Header__icon"
          >
                    <i
                              className="fa fa-shopping-cart"
                    />
          </div>
          <div
                    className="Header__value__text"
          >
                    <p>
                              Total
                    </p>
                    <p>
                              R$ 
                              0,00
                    </p>
          </div>
</div>,
        "className": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Header__icon"
>
              <i
                            className="fa fa-shopping-cart"
              />
</div>,
            <div
              className="Header__value__text"
>
              <p>
                            Total
              </p>
              <p>
                            R$ 
                            0,00
              </p>
</div>,
          ],
          "className": "Header__value",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-shopping-cart"
/>,
              "className": "Header__icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-shopping-cart",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Total
</p>,
                <p>
                  R$ 
                  0,00
</p>,
              ],
              "className": "Header__value__text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Total",
                },
                "ref": null,
                "rendered": "Total",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "R$ ",
                    "0,00",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "R$ ",
                  "0,00",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
